Z64Utils/Forms/ObjectAnalyzerForm.cs

```diff
@@ -244,19 +244,44 @@ private void listView_map_SelectedIndexChanged(object sender, EventArgs e)
                         StringWriter sw = new StringWriter();
                         for (int n = 0; n < matrices.Matrices.Count; n++)
                         {
-                            sw.WriteLine($" ┌                                                ┐ ");
+                            string[,] mtxStr = new string[4, 4];
+                            int[] columnsWidth = { 0, 0, 0, 0 };
+                            for (int i = 0; i < 4; i++)
+                            {
+                                for (int j = 0; j < 4; j++)
+                                {
+                                    // see Matrix_MtxToMtxF
+                                    int elemIndex = 4 * j + i;
+                                    byte[] buffer = matrices.Matrices[n].GetBuffer();
+                                    byte[] bytes = {
+                                        // integer part
+                                        buffer[2 * elemIndex], buffer[2 * elemIndex + 1],
+                                        // fractional part
+                                        buffer[2 * (4 * 4 + elemIndex)], buffer[2 * (4 * 4 + elemIndex) + 1],
+                                    };
+                                    float val = ArrayUtil.ReadInt32BE(bytes) / (float)0x10000;
+                                    string valStr = $"{val}";

```

Comment:
In decimal,
the smallest positive s16.16 number is `1/0x10000 = 0.0000152587890625`
the smallest s16.16 number is `0x80000000/0x10000 = -32768`
and the largest is `0x7FFFFFFF/0x10000 = 32767.9999847412109375`

so that's 5.16f maybe? and I don't know how it works in C# either. Microsoft probably thought it was funny to make up [their own](https://xkcd.com/927/) format strings

Reply:
Rereading this,

> the smallest positive s16.16 number is `1/0x10000 = 0.0000152587890625`

absolutely doesn't mean we need 16 digits, but only 5

idk still about this, it's not like there isn't enough space in Z64Utils to show whatever useless digits